internal/types/host/keepalived_host_collector_host.go

```diff
@@ -0,0 +1,270 @@
+package host
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"github.com/sirupsen/logrus"
+
+	"github.com/cafebazaar/keepalived-exporter/internal/collector"
+	"github.com/cafebazaar/keepalived-exporter/internal/types/utils"
+)
+
+var sigNumSupportedVersion = version.Must(version.NewVersion("1.3.8"))
+
+// KeepalivedHostCollectorHost implements Collector for when Keepalived and Keepalived Exporter are both on a same host
+type KeepalivedHostCollectorHost struct {
+	pidPath string
+	version *version.Version
+	useJSON bool
+
+	SIGJSON  syscall.Signal
+	SIGDATA  syscall.Signal
+	SIGSTATS syscall.Signal
+}
+
+// NewKeepalivedHostCollectorHost is creating new instance of KeepalivedHostCollectorHost
+func NewKeepalivedHostCollectorHost(useJSON bool, pidPath string) *KeepalivedHostCollectorHost {
+	k := &KeepalivedHostCollectorHost{
+		useJSON: useJSON,
+		pidPath: pidPath,
+	}
+
+	var err error
+	k.version, err = k.getKeepalivedVersion()
+	if err != nil {
+		logrus.WithError(err).Warn("Version detection failed. Assuming it's the latest one.")
+	}
+
+	k.getSignals()
+
+	return k
+}
+
+func (k *KeepalivedHostCollectorHost) getSignals() {
+	if k.useJSON {
+		k.SIGJSON = k.sigNum("JSON")
+	}
+	k.SIGDATA = k.sigNum("DATA")
+	k.SIGSTATS = k.sigNum("STATS")
+}
+
+// GetKeepalivedVersion returns Keepalived version
+func (k *KeepalivedHostCollectorHost) getKeepalivedVersion() (*version.Version, error) {
+	cmd := exec.Command("bash", "-c", "keepalived -v")
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		logrus.WithFields(logrus.Fields{"stderr": stderr.String(), "stdout": stdout.String()}).WithError(err).Error("Error getting keepalived version")
+		return nil, errors.New("Error getting keepalived version")
+	}
+
+	version, err := utils.ParseVersion(stdout.String())
```

Comment:
Errors all log in `ParseVersion` function. Does it need to log here again?

```suggestion
	return utils.ParseVersion(stdout.String())
```

Reply:
Why not. It makes sense.